homework-2/middewares/authMiddlewares.ts

```diff
@@ -0,0 +1,40 @@
+import { NextFunction, Request, Response } from "express";
+import { sign, verify } from "jsonwebtoken"
+import httpStatus from "http-status";
+import { HttpError } from "../errors/HttpError";
+import UserService  from '../services/userService';
+
+interface IAuthReqBody {
+    login: string;
+    password: string;
+}
+
+const { JWT_SECRET } = process.env;
+
+export async function login(req: Request<unknown, unknown, IAuthReqBody>, res: Response, next: NextFunction): Promise<void> {
+    const { login, password } = req.body;
+    const user = await UserService.getUserByLogin(login);
+
+    if (user === undefined || user.password !== password || user.isDeleted) {
```

Comment:
What do you think to create specific request to DB that would check this variables?

Reply:
Good idea, I have implemented in `getAuthenticatedUser` function.